### ISSUE TITLE
ports requirements for Monitoring and other optional Rancher features

### ISSUE THREAD

**Issue by Tejeev** [rancher/docs#2493]
I couldn't find a note of the port 9796 needing to be exposed in the Rancher docs, or if any others are required.  There is conflicting info online and it looks like some ports can be modified (and may have been since thxcode notes in https://github.com/rancher/rancher/issues/17957 that 9100 is required but I see nothing expose there on my cluster).
It is important that we have this information readily available. 

I've created PRs https://github.com/rancher/docs/pull/2489 and https://github.com/rancher/docs/pull/2490 to expose at least this port on the monitoring page, but we need to find out if others are required and probably expose the information in a better way.  

I think it maybe it makes sense to add to https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/node-requirements/#networking-requirements/ and maintain it as our central reference page?  (I think that's the goal here?  Can't actually see most of the items on this page so opened #2492)

**Comment by catherineluse** [comment]
@aiyengar2 Do you know of any undocumented ports that need to be opened for monitoring? if not I think we can close this issue.

**Comment by aiyengar2** [comment]
@Tejeev I'd need to test it out to make sure, but I'm pretty sure we don't need any open ports on (non-Windows, Windows is tbd) clusters anymore in Monitoring V2. 

Monitoring V1 still has these port requirements since it uses the upstream Prometheus exporters, but Monitoring V2 uses [`rancher-pushprox`](https://github.com/rancher/charts/tree/main/charts/rancher-pushprox), which takes care of this requirement by deploying a client onto each node to collect the metrics from the hostNetwork and expose them on the Pod rather than requiring the port to be externally accessible; this was implemented for kubeScheduler, kubeProxy, kubeControllerManager, and kubeEtcd and can easily be extended to node-exporter (although, to my knowledge, node-exporter already exposes metrics via Pod rather than via Host as well).

For Windows nodes, 9796 will still be required at the moment since there's no way for us to currently publish a port on the hostNetwork of the Windows node (as exposed by wmi_exporter) via a container. However, the story for Monitoring V2 Windows support is still being worked on (https://github.com/rancher/rancher/issues/28327), so it's possible this requirement will disappear too.

**Comment by aiyengar2** [comment]
@catherineluse maybe we should keep this issue open until we test to make sure Monitoring is deployable on a cluster with locked down ports.

@deniseschannon  @SheilaghM @cbron @jiaqiluo could we prioritize testing this?

**Comment by catherineluse** [comment]
Ok, I'll leave it open.

Note: Bastian reorganized the port requirements (https://github.com/rancher/docs/pull/2637), so any pull requests should be made against the staging branch.

**Comment by Tejeev** [comment]
Monitoring v1 is still the most used and will be for a while going forward (until v2 can match the v1 feature set in the UI).  Can we be sure the v1 docs don't miss anything?

**Comment by dbason** [comment]
Piggybacking on to this issue for a support case I've just been looking at.  We also need to include 10254 when ingress-nginx is using hostNetwork, and 10250. Prometheus is trying to scrape nginx, and the cadvisor endpoint for the kubelets.

**Comment by dbason** [comment]
I've dug a little further on the ports.  The endpoints that v1 tries to scrape are controlled by headless services in the cattle-prometheus namespace.  So in addition to the ports on the workers we also need to open up the kubernetes components ports on the controlplane, and the etcd port on etcd nodes.

**Comment by dbason** [comment]
@aiyengar2 I've reviewed the monitoring v2 and we will still need 9796 (node-exporter) and 10250 (kubelet) on the workers.  Should I raise an RFE and work on the helm chart to set up a push proxy for those components?

9796 is required because the node-exporter runs on the host network.  I'm not sure if this is required for network stats gathering or not.  Interestingly the kubelet ServiceMonitor is in the kube-system namespace not cattle-monitoring-system like the rest of the components.

**Comment by aiyengar2** [comment]
Yes, let's raise an RFE for that. Adding a pushprox sounds like the right solution.

re: kubelet ServiceMonitor being in the kube-system namespace, this is because the Service that we monitor also exists in that namespace. To support users who enable flags like [ignoreNamespaceSelectors](https://github.com/rancher/charts/blob/5c4286072a01c8ce3eb45bbfb5dfb011c496d138/charts/rancher-monitoring/rancher-monitoring/9.4.203/values.yaml#L2105), the ServiceMonitor needs to exist in the same namespace as well.

**Comment by dbason** [comment]
Don't we create that service as part of the chart though?  Couldn't we put both in cattle-monitoring-system to keep all of the k8s components together?

I'll raise the RFE.

**Comment by aiyengar2** [comment]
We don't create that service as part of the chart
